### PR TITLE
EWSUtilities.getXSDurationToTimeSpan fix

### LINKED ISSUE #357: How to use setStartTimeZone method of Appointment class?
Hi All,
Can anyone share the code-snippets of using setStartTimeZone() method of Appointment class.

### PR DESCRIPTION
**Pull request by itbw**
* This pull request fixes the problem with wrong parsing of Bias in Period. 

* Fixes  #357, #398 and #343 

* Problem: 
 * TimeSpan for this one -PT3H30M will be parsed as:
```
negative true
year 0
month 30
day 0
hour 3
minute 30
seconds 0
```
* The big M for months and minutes seems to be correct according to ISO 8601 and the only difference is 'T' in the middle. For example this one P1Y2M10DT2H30M is a correct Time interval.

* Fix:
 * Two method are refactored getXSDurationToTimeSpan() and getXSDurationToTimeSpanValue().
 * For parsing of TimeSpan, an Joda-Time Period is now used instead of row Java Regex.
 * Period.parse() can automatically parse any value in form ```P1Y2M10DT2H30M```, except of leading '-' symbol. For parsing '-' Java Regex is still used. 

* Test
 * Corresponding tests are added to new class TimeSpanUtilsTest

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@itbw__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java [hunk 1]
```diff
@@ -875,79 +870,50 @@ public static TimeSpan getXSDurationToTimeSpan(String xsDuration) {
...
+    debugMessage.append(" / ");
+    debugMessage.append("hour: " + hour);
+    debugMessage.append(" / ");
+    debugMessage.append("minute: " + minute);
+    debugMessage.append(" / ");
+    debugMessage.append("seconds: " + seconds);
+    debugMessage.append(" / ");
+    debugMessage.append("milliseconds: " + milliseconds);
+
+    LOG.debug("Parded TimeSpan: " + debugMessage.toString());
```
Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java]: this seems to be a lot overhead for a debug message?!

File: src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java [hunk 2]
```diff
@@ -875,79 +870,50 @@ public static TimeSpan getXSDurationToTimeSpan(String xsDuration) {
...
+    long retval =
+        day * TimeSpan.DAYS + hour * TimeSpan.HOURS + minute * TimeSpan.MINUTES + seconds
+            * TimeSpan.SECONDS + milliseconds * TimeSpan.MILLISECONDS;
+    
     if (negative) {
       retval = -retval;
     }
-    return new TimeSpan(retval);
 
+    return new TimeSpan(retval);
```
Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java]: as TimeSpan is initialized with milliseconds here you can also use:
`period.toStandardDuration().getMillis()` rather then computing everything on your own.

Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java]: ref. [Period.html#toStandardDuration](http://joda-time.sourceforge.net/apidocs/org/joda/time/Period.html#toStandardDuration())

File: src/test/java/microsoft/exchange/webservices/data/util/TimeSpanUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  // Tests for EwsUtilities.getXSDurationToTimeSpan()
+
+  private static final String PERIOD_HOURS = "-PT13H";
+  private static final String PERIOD_HOURS_MINUTES = "-PT5H30M";
+  private static final String PERIOD_FULL = "PT2H30M59.0S";
+  private static final String PERIOD_FULL_NEGATIVE = "-PT2H30M59.0S";
+
+
+  @Test
+  public void testPeriodHours() {
```
Comment (serious6) [inline src/test/java/microsoft/exchange/webservices/data/util/TimeSpanUtilsTest.java]: maybe also add some negative tests (illegal ones), any tests for boundary values?

File: src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java (hunk 2 above)
Comment (itbw) [inline src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java]: @serious6 I had a closer at Joda Time Period. It seems, that it does not support duration with years and months, which makes sense, because year and months have different length. Here we have two methods: getXSDurationToTimeSpan(), which tries to include years and months, and getXSDurationToTimeSpanValue(), which is not. 
Is there any particular reason for having two methods or I can just merge leave one method getXSDurationToTimeSpanValue() and refactor usage of getXSDurationToTimeSpan() in other classes?
